app/admin/components/post/SubmitPostForm.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { Form, Button, Message } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import './styles.scss';
+
+const SubmitPostForm = ({
+  onSubmit, onChange, errors, internshipDetails,
+}) => (
+  <div className="submit-post-form-container">
+    {_.has(errors, 'summary') &&
+      <Message
+        error
+        content={errors.summary}
+      />
+    }
+    <Form onSubmit={onSubmit}>
+      <Form.Field>
+        <label htmlFor="submit-post-position">Position</label>
+        <input
+          name="position"
+          value={internshipDetails.name}
+          onChange={onChange}
+          id="submit-post-position"
+          placeholder="Enter the internship position..."
+        />
+      </Form.Field>
+      <Form.Field>
+        <label htmlFor="submit-post-company">Company</label>
+        <input
+          name="company"
+          value={internshipDetails.company}
+          onChange={onChange}
+          id="submit-post-company"
+          placeholder="Enter the company..."
+        />
+      </Form.Field>
+      <Form.Field>
+        <label htmlFor="submit-post-location">Location</label>
+        <input
+          name="location"
+          value={internshipDetails.location}
+          onChange={onChange}
+          id="submit-post-location"
+          placeholder="Enter the location..."
+        />
+      </Form.Field>
+      <Form.Group>
+        <Form.Field>
+          <label htmlFor="submit-post-start-date">Start Date</label>
+          <input
+            name="start-date"
+            value={internshipDetails.startDate}
+            onChange={onChange}
+            id="submit-post-start-date"
+            type="date"
+            placeholder="Enter the starting date.."
+          />
+        </Form.Field>
+        <Form.Field>
+          <label htmlFor="submit-post-duration">Duration</label>
+          <input
+            name="duration"
+            value={internshipDetails.duration}
+            onChange={onChange}
+            id="submit-post-location"
+            type="number"
+            placeholder="Enter the number of months..."
```

Comment:
let the type be string so without that increment option [ so that unit of  duration can be mentioned]